Intermittent test failure: TestNimbusExperimentBySlugQuery.test_feature_config_with_multiple_features

Seen here:

https://app.circleci.com/pipelines/github/mozilla/experimenter/25838/workflows/1111e439-5026-4bb6-bd11-6b653edeb767/jobs/92179

```py
=================================== FAILURES ===================================
__ TestNimbusExperimentBySlugQuery.test_feature_config_with_multiple_features __
[gw2] linux -- Python 3.11.1 /usr/local/bin/python3.11

self = <experimenter.experiments.tests.api.v5.test_queries.TestNimbusExperimentBySlugQuery testMethod=test_feature_config_with_multiple_features>

    def test_feature_config_with_multiple_features(self):
        user_email = "user@example.com"
        feature_config1 = NimbusFeatureConfigFactory.create(
            slug="a", application=NimbusExperiment.Application.DESKTOP
        )
        feature_config2 = NimbusFeatureConfigFactory.create(
            slug="b", application=NimbusExperiment.Application.DESKTOP
        )
    
        experiment = NimbusExperimentFactory.create_with_lifecycle(
            NimbusExperimentFactory.Lifecycles.CREATED,
            application=NimbusExperiment.Application.DESKTOP,
            feature_configs=[feature_config1, feature_config2],
        )
    
        response = self.query(
            """
            query experimentBySlug($slug: String!) {
                experimentBySlug(slug: $slug) {
                    featureConfigs {
                        id
                        application
                    }
                }
            }
            """,
            variables={"slug": experiment.slug},
            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
        )
        self.assertEqual(response.status_code, 200, response.content)
        content = json.loads(response.content)
        experiment_data = content["data"]["experimentBySlug"]
>       self.assertEqual(
            experiment_data["featureConfigs"],
            [
                {"id": feature_config1.id, "application": "DESKTOP"},
                {"id": feature_config2.id, "application": "DESKTOP"},
            ],
        )
E       AssertionError: Lists differ: [{'id': 120, 'application': 'DESKTOP'}, {'id': 119, 'application': 'DESKTOP'}] != [{'id': 119, 'application': 'DESKTOP'}, {'id': 120, 'application': 'DESKTOP'}]
E       
E       First differing element 0:
E       {'id': 120, 'application': 'DESKTOP'}
E       {'id': 119, 'application': 'DESKTOP'}
E       
E       - [{'application': 'DESKTOP', 'id': 120}, {'application': 'DESKTOP', 'id': 119}]
E       ?                                    ^^                                     ^^
E       
E       + [{'application': 'DESKTOP', 'id': 119}, {'application': 'DESKTOP', 'id': 120}]
E       ?                                    ^^                                     ^^

experimenter/experiments/tests/api/v5/test_queries.py:1740: AssertionError

---------- coverage: platform linux, python 3.11.1-final-0 -----------
```

This is a classic ordering error.  Here we'd want to compare sets but you can't put dicts in sets so another way to do it is to assert the length and then assert membership.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3642)

--- a/experimenter/experimenter/experiments/tests/api/v5/test_queries.py
+++ b/experimenter/experimenter/experiments/tests/api/v5/test_queries.py
@@ -1702,7 +1702,12 @@ class TestNimbusExperimentBySlugQuery(GraphQLTestCase):
         feature_config = experiment.feature_configs.get()
         self.assertEqual(
             experiment_data["featureConfigs"],
-            [{"id": feature_config.id, "application": "DESKTOP"}],
+            [
+                {
+                    "id": feature_config.id,
+                    "application": NimbusExperiment.Application.DESKTOP.name,
+                }
+            ],
         )
 
     def test_feature_config_with_multiple_features(self):
@@ -1737,12 +1742,20 @@ class TestNimbusExperimentBySlugQuery(GraphQLTestCase):
         self.assertEqual(response.status_code, 200, response.content)
         content = json.loads(response.content)
         experiment_data = content["data"]["experimentBySlug"]
-        self.assertEqual(
+        self.assertEqual(len(experiment_data["featureConfigs"]), 2)
+        self.assertIn(
+            {
+                "id": feature_config1.id,
+                "application": NimbusExperiment.Application.DESKTOP.name,
+            },
             experiment_data["featureConfigs"],
-            [
-                {"id": feature_config1.id, "application": "DESKTOP"},
-                {"id": feature_config2.id, "application": "DESKTOP"},
-            ],
+        )
+        self.assertIn(
+            {
+                "id": feature_config2.id,
+                "application": NimbusExperiment.Application.DESKTOP.name,
+            },
+            experiment_data["featureConfigs"],
         )
 
     def test_branches(self):

